emcc.py

```diff
@@ -97,6 +97,8 @@
     '-rpath': True,
     '-rpath-link': True,
     '-version-script': True,
+    '-arch': True,
```

Comment:
This one is very worrying.  if the build system is passing `-arch=arm64` to emcc it shows that something is badly misconfigured.  I don't think we should ignore this one.

Reply:
agreed, but it can happen in context of cross-compilation with python right now pretty easily ...
The issue is that python - as it is right now - belongs to the host.  Of course you can override the sysconfigdata by the export PYTHON_SYSCONFIGDATA_NAME parts of it.  But some stuff remains, like 'sys.platform', or 'platform.uname'.  So use of setuptools, pip, etc might still see the operating system and selects "correct" compiler flags

Reply:
I don't understand.  Can you explain more about why python pass `-arch=arm64`.. is it trying to build some utility for the host system?  If so, it should not be using emcc at all right?   What does this have to do with `PYTHON_SYSCONFIGDATA_NAME`?

If python doesn't have a good cross compilation story it should really be fixed on the python side, shouldn't it?  Not by hacking emcc like this?

Reply:
Python sadly hasn't any good cross-compilation story right now.  Parts of it can be work-a-rounded by redirecting to a specific sysconfigdata file to be used (environment variable PYTHON_SYSCONFIGDATA_NAME).  But there are other information, which are currently hard-coded in core python, which can't be modified as easy in a reliable way.
So the issue about '-arch arm64' - as sample - comes from setup.py (or equivalent).

Btw the same can happen if make-system relies on uname funtionionalities to gather information on the host, which could lead to such "additional" options too.

Reply:
anyway, agreed that we could remove -arch here as to be ignored option